lib/is-semantic-message.js

```diff
@@ -17,6 +17,8 @@ module.exports = function isSemanticMessage (message, validScopes, validTypes, a
 
   const [result] = commits
   const { scope, type } = result.header
-  const isScopeValid = !validScopes || !scope || validScopes.includes(scope)
+  const isScopeValid = !validScopes || !scope || validScopes.some(function(valid) { 
+    return new RegExp(valid.replace(/(^\/|\/$)/g, "")).test(scope) 
```

Comment:
Curious if there's a way to support this without requiring this `replace` call, as it makes some assumptions about the format of the incoming regex...

Reply:
I see your point. The thought was to differentiate in the yaml between a regex and a vanilla string. 

Instead the scope could just accept and run everything as a regex test. All of these would be valid scopes: 
```yaml
# You can define a list of valid scopes
scopes:
  - scope1
  - ^TASK-(\\d)+$
  - scope\\d
  ...
```

Then the function would work like this:
```ts
const isScopeValid = !validScopes || !scope || validScopes.some(function(valid) { 
    return new RegExp(valid).test(scope) 
}
```

Would that be better?

Reply:
We need some way to know to match as a regex or an exact match. 

The issue is if everything ran as a regex then we'd be accepting substring.
`new RegExp('validScope').test('invalidScope') => true (but should fail)`
`'validScope' === 'invalidScope'  => false (which is correct)`

I assumed if the scope starts & ends as `/` that's would be a good way to know if we run as an exact match or a regex. Either `scope.replace(isRegExp)` or `scope[0] === "/" && scope[scope.length-1] === "/"` are valid test but we would need to remove the first & last slash before we send to the RegExp function.